### PR TITLE
man: describe the relationship between nss-myhostname and nss-resolved

### LINKED ISSUE #1605: Document intersection of nss-myhostname and nss-resolve
While testing out nss-resolve, I discovered that after enabling it, myhostname becomes useless:
- *.localhost maps to 127.0.0.1 and ::1
- The local hostname resolves to the public addresses of the host
- gateway resolves to the routing gateway address

All three tested with `getent ahosts $1`, having disabled myhostname. Has nss-myhostname become superfluous when nss-resolve is installed? Or is this result only sometimes applicable? The documentation doesn't say, and in particular `nss-resolve(8)` does not mention that one should expect this behavior.

systemd 227-2 from debian (plus enabling nss-resolve).

### PR DESCRIPTION
**Pull request by keszybz**
Fixes #1605.

### CODE REVIEW

File: man/systemd.service.xml [hunk 1]
```diff
@@ -916,12 +916,14 @@
 
     <para>Each command line is split on whitespace, with the first item being the command to
     execute, and the subsequent items being the arguments. Double quotes ("…") and single quotes
-    ('…') may be used, in which case everything until the next matching quote becomes part of the
-    same argument. Quotes themselves are removed. C-style escapes are also supported. The table
-    below contains the list of known escape patterns. Only escape patterns which match the syntax in
-    the table are allowed; other patterns may be added in the future and unknown patterns will
-    result in a warning. In particular, any backslashes should be doubled. Finally, a trailing
-    backslash (<literal>\</literal>) may be used to merge lines.</para>
+    ('…') may be used to wrap a whole item (the opening quote may appear only after whitespace, and
```
Comment (floppym) [inline man/systemd.service.xml]: > the opening quote may appear only after whitespace

That's not quite true. An opening quote may also appear immediately following `ExecXXX=`.
```
ExecStart="/path with spaces/mydaemon" ...
```

**Comment by keszybz** [comment]
Updated.

**Comment by keszybz** [comment]
@Conan-Kudo: please have a look at the last patch.

File: man/systemd-nspawn.xml [hunk 2]
```diff
@@ -1091,6 +1091,17 @@
     </example>
 
     <example>
+      <title>Install the OpenSUSE Tumbleweed rolling distribution</title>
+
+      <programlisting># zypper --root=/var/lib/machines/tumbleweed ar \
+      https://download.opensuse.org/tumbleweed/repo/oss tumbleweed
+# zypper --root=/var/lib/machines/tumbleweed refresh
+# zypper --root=/var/lib/machines/tumbleweed install --no-recommends systemd
```
Comment (Conan-Kudo) [inline man/systemd-nspawn.xml]: The package list should be `systemd shadow zypper openSUSE-release vim`, I believe.

File: man/systemd-nspawn.xml [hunk 3]
```diff
@@ -1091,6 +1091,17 @@
     </example>
 
     <example>
+      <title>Install the OpenSUSE Tumbleweed rolling distribution</title>
+
+      <programlisting># zypper --root=/var/lib/machines/tumbleweed ar \
```
Comment (Conan-Kudo) [inline man/systemd-nspawn.xml]: Add `-t yast2`, or it may not detect the repo properly...

Comment (Conan-Kudo) [inline man/systemd-nspawn.xml]: Add `-c` after `ar`, or it may not detect the repo properly...

**Comment by Conan-Kudo** [comment]
@keszybz See feedback above.

**Comment by keszybz** [comment]
Thanks, updated.

File: man/systemd-nspawn.xml [hunk 4]
```diff
@@ -1132,6 +1144,7 @@
       <citerefentry project='mankier'><refentrytitle>dnf</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
       <citerefentry project='die-net'><refentrytitle>debootstrap</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
       <citerefentry project='archlinux'><refentrytitle>pacman</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <ulink url="https://en.opensuse.org/SDB:Zypper_usage">Zypper</ulink>,
```
Comment (Conan-Kudo) [inline man/systemd-nspawn.xml]: I'd suggest using [the Zypper man page](https://www.mankier.com/8/zypper) instead. The wiki page is out of date...

**Comment by keszybz** [comment]
> I'd suggest using [the Zypper man page](https://www.mankier.com/8/zypper) instead.

Yes, that's better. Updated again.

File: man/nss-resolve.xml [hunk 5]
```diff
@@ -63,13 +63,24 @@
...
+    <literal>hosts:</literal> in <filename>/etc/nsswitch.conf</filename>. Specifically, it is recommended to place
+    <literal>resolve</literal> early in <filename>/etc/nsswitch.conf</filename>'s <literal>hosts:</literal> line (but
     after the <literal>files</literal> or <literal>mymachines</literal> entries), right before the
     <literal>dns</literal> entry if it exists, followed by <literal>[!UNAVAIL=return]</literal>, to ensure DNS queries
     are always routed via
     <citerefentry><refentrytitle>systemd-resolved</refentrytitle><manvolnum>8</manvolnum></citerefentry> if it is
     running, but are routed to <command>nss-dns</command> if this service is not available.</para>
+
+    <para>Note that <command>systemd-resolved</command> will synthesize DNS resource
+    records in a few cases, for example for the <literal>localhost</literal> and the
```
Comment (poettering) [inline man/nss-resolve.xml:75]: I would have just written "localhost" instead of "the localhost", after all, this is an established concept.

**Pull request merged**